### PR TITLE
Account service implementation

### PR DESCRIPTION
**Pull request by NiazBikbaev**

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class AccountServiceImpl implements AccountService {
+
+    private static Logger logger = LoggerFactory.getLogger(AccountServiceImpl.class.getName());
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: We have a Lombok on classpath, use a `@Sl4j` annotation on class and Lombok will kindly generate and supply a `log` field with logger for you

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Autowired
+    private AccountRepository accountRepository;
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Avoid field injection, use constructor injection
You can avoid explicit constructor either
```java
@RequiredArgsConstructor
public class AccountServiceImpl implements AccountService {

private final AccountRepository accountRepository;
//...
}
```
Lombok will generate a constructor for your `final` fields and Spring is smart enough to inject them, try it

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    @Transactional
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: There is a single call to `accountRepository.save` no other database related actions are not happening, why `@Transactional` ?

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    public Account registerAccount(Account account) {
+        Account savedAccount = null;
+        try {
+            savedAccount = accountRepository.save(account);
+        } catch (DataAccessException exception) {
+            logger.warn(exception.getMessage(), exception);
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: is it really a level 'warn' exception when service is failing to add a new account?

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java (hunk 1 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Ok

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java (hunk 2 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Ok

File: src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java (hunk 3 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/AccountServiceImpl.java]: Before that I tested several operations, just forgot to remove it

**Pull request merged**